### PR TITLE
Moar docs

### LINKED ISSUE #64: Test examples in README
If we converted the examples in the README to doctests, we could use https://pytest.org/latest/doctest.html.

Some extra work will be needed to fake the access tokens.

### PR DESCRIPTION
**Pull request by perrygeo**
This is a 30% design, just putting it out there to get some feedback.

This PR is focused on docs, docs and only docs

* Add markdown file for each service with doctests
* Removes code examples from the readme and instead links to the appropriate doc
* Pulls out some common doc topics (access tokens, input features, etc)

There is still work to do:
* [x] use token with upload scope
* [x] review for errors, language, organization, consistency
* [x] cover some of the additional options, kwargs and other use cases
* [x] Upload API doctests fail because they require a username in the constructor
* [x] Keep code in interpreter blocks to 72-79 chars wide
* [x] Make assertions on headers to demonstrate content types

resolves #64

### CODE REVIEW

File: docs/geocoding.md [hunk 1]
```diff
@@ -14,52 +13,27 @@ access to the Mapbox Geocoding API. You can also import it directly from the
 See https://www.mapbox.com/developers/api/geocoding/ for general documentation
 of the API.
 
-# Access Tokens
+Your Mapbox access token should be set in your environment; eee the [access tokens](access_tokens.md) documentation for more information.
```
Comment (sgillies) [inline docs/geocoding.md]: s/eee/see/

:+1: on moving access tokens to its own file.

File: docs/static.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+>>> service = Static()
+
+```
+
+```python
+>>> response = service.image('mapbox.satellite',
+...                          lon=-61.7, lat=12.1, z=12)
+
+>>> response.content[1:4] == 'PNG'
+True
```
Comment (sgillies) [inline docs/static.md]: @perrygeo I think more instructive would be to show

```python
>>> r.headers['Content-Type']
'image/png'
```

**Comment by sgillies** [comment]
Totally on the right track. I'm adding a couple other checkboxes to the top comment.

**Comment by perrygeo** [comment]
I'm skipping the doctests for the upload functionality - no need to block docs as it will be handled in #86

**Comment by sgillies** [comment]
From above:

> * [ ] still some repetition at the beginning of each service's doc, could probably refactor into topic docs

Let's tackle this in a future PR and land this sucker.

**Comment by perrygeo** [comment]
:+1:

**Pull request merged**